webhooks/models.py

```diff
@@ -1,27 +1,30 @@
-import json
-
 from typing_extensions import Literal
-from pydantic import ValidationError
+from aries_cloudcontroller import (
+    ConnRecord,
+    V10PresentationExchange,
+    V20PresExRecord,
+    V10CredentialExchange,
+    V20CredExRecord,
+)
 
 from shared_models import (
-    Connection,
+    ConnectionsHook,
+    credential_record_to_model_v1,
     PresentationExchange,
     CredentialExchange,
     HookBase,
+    presentation_record_to_model,
+    conn_record_to_connection,
+    credential_record_to_model_v2,
 )
 
 
-class ConnectionsHook(HookBase, Connection):
-    pass
-
-
 def to_connections_model(item: dict) -> ConnectionsHook:
-    item["state"] = item.pop("rfc23_state")
-    try:
-        item = ConnectionsHook(**item)
-    except ValidationError:
-        pass
-    return item
+    conn_record = ConnRecord(**item)
+    conn_record = conn_record_to_connection(connection_record=conn_record)
+
+    merged_dicts = {**item, **conn_record.dict()}
```

Comment:
Why are both the plain dict and the conn record merged here?

Reply:
because the item has eg the wallet_id and oriigin fields which the conn record doesnt have.

The way this works is that the left-most gets used first (including values we dont want and have overwritten (eg moved '_' to '-')BUT then the conn record/item on the right gets used to update and override existing fields (with the values we have chanhed to the model. In essence the all the fields of the model we use get overridden the way we want and the ones we dont have in the model but specific from the webhook event come from item

And yes, they now match 100%

Reply:
example:

```python
a = {'a': 1, 'b': 2, 'c':3}
b = {'b': 3, 'c':5}
c = {**a, **b}
print(c)
# gives
{'a': 1, 'b': 3, 'c':5}
```